{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 폴더 이름 변경: ../../data/labels_det -> ../../data/labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import numpy as np\n",
    "# # save np.load\n",
    "# np_load_old = np.load\n",
    "\n",
    "# # modify the default parameters of np.load\n",
    "# np.load = lambda *a,**k: np_load_old(*a, allow_pickle=True, **k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ultralytics import YOLO\n",
    "\n",
    "# Load a model\n",
    "# model = YOLO('yolov8n.yaml')  # build a new model from YAML\n",
    "model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)\n",
    "# model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "New https://pypi.org/project/ultralytics/8.0.154 available  Update with 'pip install -U ultralytics'\n",
      "Ultralytics YOLOv8.0.152  Python-3.9.17 torch-2.0.1+cpu CPU (12th Gen Intel Core(TM) i3-12100F)\n",
      "\u001b[34m\u001b[1mengine\\trainer: \u001b[0mtask=detect, mode=train, model=yolov8n.pt, data=datadet.yml, epochs=100, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs\\detect\\train4\n",
      "Overriding model.yaml nc=80 with nc=23\n",
      "\n",
      "                   from  n    params  module                                       arguments                     \n",
      "  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 \n",
      "  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                \n",
      "  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             \n",
      "  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                \n",
      "  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             \n",
      "  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               \n",
      "  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           \n",
      "  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              \n",
      "  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           \n",
      "  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 \n",
      " 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n",
      " 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 \n",
      " 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n",
      " 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  \n",
      " 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                \n",
      " 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 \n",
      " 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              \n",
      " 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 \n",
      " 22        [15, 18, 21]  1    755797  ultralytics.nn.modules.head.Detect           [23, [64, 128, 256]]          \n",
      "Model summary: 225 layers, 3015333 parameters, 3015317 gradients\n",
      "\n",
      "Transferred 319/355 items from pretrained weights\n",
      "\u001b[34m\u001b[1mTensorBoard: \u001b[0mStart with 'tensorboard --logdir runs\\detect\\train4', view at http://localhost:6006/\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mScanning C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\data\\labels\\train... 45139 images, 0 backgrounds, 1 corrupt: 100%|██████████| 45139/45139 [00:48<00:00, 925.59it/s] \n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\data\\images\\train\\C000895_000_0090_C_D_F_0.jpg: 1 duplicate labels removed\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\data\\images\\train\\T014403_000_0414_B_D_F_1.jpg: 1 duplicate labels removed\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\data\\images\\train\\T015350_007_0378_C_D_F_1.jpg: ignoring corrupt image/label: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mNew cache created: C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\data\\labels\\train.cache\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\data\\labels\\val... 5642 images, 0 backgrounds, 0 corrupt: 100%|██████████| 5642/5642 [00:06<00:00, 925.08it/s]\n",
      "\u001b[34m\u001b[1mval: \u001b[0mNew cache created: C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\data\\labels\\val.cache\n",
      "Plotting labels to runs\\detect\\train4\\labels.jpg... \n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50724 (\\N{HANGUL SYLLABLE O}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53664 (\\N{HANGUL SYLLABLE TO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48148 (\\N{HANGUL SYLLABLE BA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51060 (\\N{HANGUL SYLLABLE I}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48372 (\\N{HANGUL SYLLABLE BO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54665 (\\N{HANGUL SYLLABLE HAENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51088 (\\N{HANGUL SYLLABLE JA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46020 (\\N{HANGUL SYLLABLE DO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47196 (\\N{HANGUL SYLLABLE RO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53685 (\\N{HANGUL SYLLABLE TONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50504 (\\N{HANGUL SYLLABLE AN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51204 (\\N{HANGUL SYLLABLE JEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47784 (\\N{HANGUL SYLLABLE MO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48120 (\\N{HANGUL SYLLABLE MI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52265 (\\N{HANGUL SYLLABLE CAG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50857 (\\N{HANGUL SYLLABLE YONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47924 (\\N{HANGUL SYLLABLE MU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 45800 (\\N{HANGUL SYLLABLE DAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54945 (\\N{HANGUL SYLLABLE HOENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49888 (\\N{HANGUL SYLLABLE SIN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54840 (\\N{HANGUL SYLLABLE HO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50948 (\\N{HANGUL SYLLABLE WI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48152 (\\N{HANGUL SYLLABLE BAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51221 (\\N{HANGUL SYLLABLE JEONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51648 (\\N{HANGUL SYLLABLE JI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49440 (\\N{HANGUL SYLLABLE SEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51452 (\\N{HANGUL SYLLABLE JU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 44144 (\\N{HANGUL SYLLABLE GEO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52880 (\\N{HANGUL SYLLABLE KAE}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47532 (\\N{HANGUL SYLLABLE RI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50612 (\\N{HANGUL SYLLABLE EO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53413 (\\N{HANGUL SYLLABLE KIG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46300 (\\N{HANGUL SYLLABLE DEU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46041 (\\N{HANGUL SYLLABLE DONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49849 (\\N{HANGUL SYLLABLE SEUNG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53457 (\\N{HANGUL SYLLABLE TAB}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50724 (\\N{HANGUL SYLLABLE O}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53664 (\\N{HANGUL SYLLABLE TO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48148 (\\N{HANGUL SYLLABLE BA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51060 (\\N{HANGUL SYLLABLE I}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48372 (\\N{HANGUL SYLLABLE BO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54665 (\\N{HANGUL SYLLABLE HAENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51088 (\\N{HANGUL SYLLABLE JA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46020 (\\N{HANGUL SYLLABLE DO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47196 (\\N{HANGUL SYLLABLE RO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53685 (\\N{HANGUL SYLLABLE TONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50504 (\\N{HANGUL SYLLABLE AN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51204 (\\N{HANGUL SYLLABLE JEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47784 (\\N{HANGUL SYLLABLE MO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48120 (\\N{HANGUL SYLLABLE MI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52265 (\\N{HANGUL SYLLABLE CAG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50857 (\\N{HANGUL SYLLABLE YONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47924 (\\N{HANGUL SYLLABLE MU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 45800 (\\N{HANGUL SYLLABLE DAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54945 (\\N{HANGUL SYLLABLE HOENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49888 (\\N{HANGUL SYLLABLE SIN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54840 (\\N{HANGUL SYLLABLE HO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50948 (\\N{HANGUL SYLLABLE WI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48152 (\\N{HANGUL SYLLABLE BAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51221 (\\N{HANGUL SYLLABLE JEONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51648 (\\N{HANGUL SYLLABLE JI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49440 (\\N{HANGUL SYLLABLE SEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51452 (\\N{HANGUL SYLLABLE JU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 44144 (\\N{HANGUL SYLLABLE GEO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52880 (\\N{HANGUL SYLLABLE KAE}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47532 (\\N{HANGUL SYLLABLE RI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50612 (\\N{HANGUL SYLLABLE EO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53413 (\\N{HANGUL SYLLABLE KIG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46300 (\\N{HANGUL SYLLABLE DEU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46041 (\\N{HANGUL SYLLABLE DONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49849 (\\N{HANGUL SYLLABLE SEUNG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53457 (\\N{HANGUL SYLLABLE TAB}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "\u001b[34m\u001b[1moptimizer:\u001b[0m SGD(lr=0.01, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)\n",
      "Image sizes 640 train, 640 val\n",
      "Using 0 dataloader workers\n",
      "Logging results to \u001b[1mruns\\detect\\train4\u001b[0m\n",
      "Starting training for 100 epochs...\n",
      "\n",
      "      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n",
      "      1/100         0G      1.297       3.06     0.9882          4        640: 100%|██████████| 2822/2822 [4:05:08<00:00,  5.21s/it]  \n",
      "                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 177/177 [08:45<00:00,  2.97s/it]\n",
      "                   all       5642      10004      0.448      0.412      0.338      0.228\n",
      "\n",
      "      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n",
      "      2/100         0G      1.233      2.042     0.9747         54        640:  88%|████████▊ | 2473/2822 [3:39:01<31:21,  5.39s/it]  "
     ]
    }
   ],
   "source": [
    "# Train the model\n",
    "results = model.train(data='datadet.yml', epochs=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "New https://pypi.org/project/ultralytics/8.0.154 available  Update with 'pip install -U ultralytics'\n",
      "Ultralytics YOLOv8.0.152  Python-3.9.17 torch-2.0.1+cpu CPU (12th Gen Intel Core(TM) i3-12100F)\n",
      "\u001b[34m\u001b[1mengine\\trainer: \u001b[0mtask=detect, mode=train, model=yolov8n.pt, data=datadet.yml, epochs=500, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=disk, device=0, workers=8, project=YOLOv8, name=yolov8n, exist_ok=False, pretrained=False, optimizer=SGD, verbose=False, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=False, val=True, split=val, save_json=False, save_hybrid=False, conf=0.001, iou=0.7, max_det=300, half=True, dnn=False, plots=True, source=ultralytics/assets/, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=17, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.001, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, image_weights=False, hide_labels=False, hide_conf=False, line_thickness=3, fl_gamma=0.0, v5loader=True, save_dir=YOLOv8\\yolov8n2\n",
      "\u001b[34m\u001b[1mTensorBoard: \u001b[0mStart with 'tensorboard --logdir YOLOv8\\yolov8n2', view at http://localhost:6006/\n",
      "Overriding model.yaml nc=80 with nc=23\n",
      "\n",
      "                   from  n    params  module                                       arguments                     \n",
      "  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 \n",
      "  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                \n",
      "  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             \n",
      "  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                \n",
      "  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             \n",
      "  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               \n",
      "  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           \n",
      "  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              \n",
      "  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           \n",
      "  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 \n",
      " 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n",
      " 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 \n",
      " 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n",
      " 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  \n",
      " 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                \n",
      " 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 \n",
      " 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              \n",
      " 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n",
      " 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 \n",
      " 22        [15, 18, 21]  1    755797  ultralytics.nn.modules.head.Detect           [23, [64, 128, 256]]          \n",
      "Model summary: 225 layers, 3015333 parameters, 3015317 gradients\n",
      "\n",
      "Transferred 319/355 items from pretrained weights\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mScanning C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\데이터\\labels\\train... 45139 images, 0 backgrounds, 11 corrupt: 100%|██████████| 45139/45139 [00:46<00:00, 971.57it/s] \n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T007931_000_0070_B_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0263      1.0167]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008034_001_0144_B_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.2102]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_066_0126_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0133      1.0782]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_066_0144_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0145      1.0796]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_067_0096_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0236]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_068_0040_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3539      1.2685      1.1059]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_068_0100_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3495      1.5021]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_072_0012_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3525]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_072_0156_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.354]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T008037_100_0060_C_D_F_0.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.2844       1.077]\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T014403_000_0414_B_D_F_1.jpg: 1 duplicate labels removed\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\train\\T015350_007_0378_C_D_F_1.jpg: ignoring corrupt image/label: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mNew cache created: C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\labels\\train.cache\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mCaching images (0.0GB disk): 100%|██████████| 45128/45128 [00:12<00:00, 3522.45it/s]\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\데이터\\labels\\val... 5642 images, 0 backgrounds, 0 corrupt: 100%|██████████| 5642/5642 [00:09<00:00, 564.55it/s]\n",
      "\u001b[34m\u001b[1mval: \u001b[0mWARNING  C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\images\\val\\C000895_000_0090_C_D_F_0.jpg: 1 duplicate labels removed\n",
      "\u001b[34m\u001b[1mval: \u001b[0mNew cache created: C:\\Users\\jinmo\\Documents\\GitHub\\tp2\\\\labels\\val.cache\n",
      "\u001b[34m\u001b[1mval: \u001b[0mCaching images (0.0GB disk): 100%|██████████| 5642/5642 [00:01<00:00, 4197.41it/s]\n",
      "Plotting labels to YOLOv8\\yolov8n2\\labels.jpg... \n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50724 (\\N{HANGUL SYLLABLE O}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53664 (\\N{HANGUL SYLLABLE TO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48148 (\\N{HANGUL SYLLABLE BA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51060 (\\N{HANGUL SYLLABLE I}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48372 (\\N{HANGUL SYLLABLE BO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54665 (\\N{HANGUL SYLLABLE HAENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51088 (\\N{HANGUL SYLLABLE JA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46020 (\\N{HANGUL SYLLABLE DO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47196 (\\N{HANGUL SYLLABLE RO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53685 (\\N{HANGUL SYLLABLE TONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50504 (\\N{HANGUL SYLLABLE AN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51204 (\\N{HANGUL SYLLABLE JEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47784 (\\N{HANGUL SYLLABLE MO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48120 (\\N{HANGUL SYLLABLE MI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52265 (\\N{HANGUL SYLLABLE CAG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50857 (\\N{HANGUL SYLLABLE YONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47924 (\\N{HANGUL SYLLABLE MU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 45800 (\\N{HANGUL SYLLABLE DAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54945 (\\N{HANGUL SYLLABLE HOENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49888 (\\N{HANGUL SYLLABLE SIN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54840 (\\N{HANGUL SYLLABLE HO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50948 (\\N{HANGUL SYLLABLE WI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48152 (\\N{HANGUL SYLLABLE BAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51221 (\\N{HANGUL SYLLABLE JEONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51648 (\\N{HANGUL SYLLABLE JI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49440 (\\N{HANGUL SYLLABLE SEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51452 (\\N{HANGUL SYLLABLE JU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 44144 (\\N{HANGUL SYLLABLE GEO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52880 (\\N{HANGUL SYLLABLE KAE}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47532 (\\N{HANGUL SYLLABLE RI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50612 (\\N{HANGUL SYLLABLE EO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53413 (\\N{HANGUL SYLLABLE KIG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46300 (\\N{HANGUL SYLLABLE DEU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46041 (\\N{HANGUL SYLLABLE DONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49849 (\\N{HANGUL SYLLABLE SEUNG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53457 (\\N{HANGUL SYLLABLE TAB}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50724 (\\N{HANGUL SYLLABLE O}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53664 (\\N{HANGUL SYLLABLE TO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48148 (\\N{HANGUL SYLLABLE BA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51060 (\\N{HANGUL SYLLABLE I}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48372 (\\N{HANGUL SYLLABLE BO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54665 (\\N{HANGUL SYLLABLE HAENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51088 (\\N{HANGUL SYLLABLE JA}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46020 (\\N{HANGUL SYLLABLE DO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47196 (\\N{HANGUL SYLLABLE RO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53685 (\\N{HANGUL SYLLABLE TONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50504 (\\N{HANGUL SYLLABLE AN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51204 (\\N{HANGUL SYLLABLE JEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47784 (\\N{HANGUL SYLLABLE MO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48120 (\\N{HANGUL SYLLABLE MI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52265 (\\N{HANGUL SYLLABLE CAG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50857 (\\N{HANGUL SYLLABLE YONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47924 (\\N{HANGUL SYLLABLE MU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 45800 (\\N{HANGUL SYLLABLE DAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54945 (\\N{HANGUL SYLLABLE HOENG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49888 (\\N{HANGUL SYLLABLE SIN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 54840 (\\N{HANGUL SYLLABLE HO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50948 (\\N{HANGUL SYLLABLE WI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 48152 (\\N{HANGUL SYLLABLE BAN}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51221 (\\N{HANGUL SYLLABLE JEONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51648 (\\N{HANGUL SYLLABLE JI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49440 (\\N{HANGUL SYLLABLE SEON}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 51452 (\\N{HANGUL SYLLABLE JU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 44144 (\\N{HANGUL SYLLABLE GEO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 52880 (\\N{HANGUL SYLLABLE KAE}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 47532 (\\N{HANGUL SYLLABLE RI}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 50612 (\\N{HANGUL SYLLABLE EO}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53413 (\\N{HANGUL SYLLABLE KIG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46300 (\\N{HANGUL SYLLABLE DEU}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 46041 (\\N{HANGUL SYLLABLE DONG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 49849 (\\N{HANGUL SYLLABLE SEUNG}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "c:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\utils\\plotting.py:313: UserWarning: Glyph 53457 (\\N{HANGUL SYLLABLE TAB}) missing from current font.\n",
      "  plt.savefig(fname, dpi=200)\n",
      "\u001b[34m\u001b[1moptimizer:\u001b[0m SGD(lr=0.01, momentum=0.937) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.001), 63 bias(decay=0.0)\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "yolov8n.pt training to 500 epochs is finished, nothing to resume.\nStart a new training without resuming, i.e. 'yolo train model=yolov8n.pt'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[3], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[39m# Train the model\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m results \u001b[39m=\u001b[39m model\u001b[39m.\u001b[39;49mtrain(data\u001b[39m=\u001b[39;49m\u001b[39m'\u001b[39;49m\u001b[39mdatadet.yml\u001b[39;49m\u001b[39m'\u001b[39;49m, epochs\u001b[39m=\u001b[39;49m\u001b[39m100\u001b[39;49m, resume\u001b[39m=\u001b[39;49m\u001b[39mTrue\u001b[39;49;00m)\n",
      "File \u001b[1;32mc:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\engine\\model.py:377\u001b[0m, in \u001b[0;36mModel.train\u001b[1;34m(self, trainer, **kwargs)\u001b[0m\n\u001b[0;32m    375\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mmodel \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mtrainer\u001b[39m.\u001b[39mmodel\n\u001b[0;32m    376\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mtrainer\u001b[39m.\u001b[39mhub_session \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39msession  \u001b[39m# attach optional HUB session\u001b[39;00m\n\u001b[1;32m--> 377\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mtrainer\u001b[39m.\u001b[39;49mtrain()\n\u001b[0;32m    378\u001b[0m \u001b[39m# Update model and cfg after training\u001b[39;00m\n\u001b[0;32m    379\u001b[0m \u001b[39mif\u001b[39;00m RANK \u001b[39min\u001b[39;00m (\u001b[39m-\u001b[39m\u001b[39m1\u001b[39m, \u001b[39m0\u001b[39m):\n",
      "File \u001b[1;32mc:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\engine\\trainer.py:192\u001b[0m, in \u001b[0;36mBaseTrainer.train\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    190\u001b[0m         ddp_cleanup(\u001b[39mself\u001b[39m, \u001b[39mstr\u001b[39m(file))\n\u001b[0;32m    191\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m--> 192\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_do_train(world_size)\n",
      "File \u001b[1;32mc:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\engine\\trainer.py:276\u001b[0m, in \u001b[0;36mBaseTrainer._do_train\u001b[1;34m(self, world_size)\u001b[0m\n\u001b[0;32m    273\u001b[0m \u001b[39mif\u001b[39;00m world_size \u001b[39m>\u001b[39m \u001b[39m1\u001b[39m:\n\u001b[0;32m    274\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_setup_ddp(world_size)\n\u001b[1;32m--> 276\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_setup_train(world_size)\n\u001b[0;32m    278\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mepoch_time \u001b[39m=\u001b[39m \u001b[39mNone\u001b[39;00m\n\u001b[0;32m    279\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mepoch_time_start \u001b[39m=\u001b[39m time\u001b[39m.\u001b[39mtime()\n",
      "File \u001b[1;32mc:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\engine\\trainer.py:267\u001b[0m, in \u001b[0;36mBaseTrainer._setup_train\u001b[1;34m(self, world_size)\u001b[0m\n\u001b[0;32m    265\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mscheduler \u001b[39m=\u001b[39m optim\u001b[39m.\u001b[39mlr_scheduler\u001b[39m.\u001b[39mLambdaLR(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39moptimizer, lr_lambda\u001b[39m=\u001b[39m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mlf)\n\u001b[0;32m    266\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mstopper, \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mstop \u001b[39m=\u001b[39m EarlyStopping(patience\u001b[39m=\u001b[39m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39margs\u001b[39m.\u001b[39mpatience), \u001b[39mFalse\u001b[39;00m\n\u001b[1;32m--> 267\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mresume_training(ckpt)\n\u001b[0;32m    268\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mscheduler\u001b[39m.\u001b[39mlast_epoch \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mstart_epoch \u001b[39m-\u001b[39m \u001b[39m1\u001b[39m  \u001b[39m# do not move\u001b[39;00m\n\u001b[0;32m    269\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mrun_callbacks(\u001b[39m'\u001b[39m\u001b[39mon_pretrain_routine_end\u001b[39m\u001b[39m'\u001b[39m)\n",
      "File \u001b[1;32mc:\\Users\\jinmo\\anaconda3\\envs\\tp2\\lib\\site-packages\\ultralytics\\engine\\trainer.py:592\u001b[0m, in \u001b[0;36mBaseTrainer.resume_training\u001b[1;34m(self, ckpt)\u001b[0m\n\u001b[0;32m    590\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mema\u001b[39m.\u001b[39mupdates \u001b[39m=\u001b[39m ckpt[\u001b[39m'\u001b[39m\u001b[39mupdates\u001b[39m\u001b[39m'\u001b[39m]\n\u001b[0;32m    591\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mresume:\n\u001b[1;32m--> 592\u001b[0m     \u001b[39massert\u001b[39;00m start_epoch \u001b[39m>\u001b[39m \u001b[39m0\u001b[39m, \\\n\u001b[0;32m    593\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m'\u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39margs\u001b[39m.\u001b[39mmodel\u001b[39m}\u001b[39;00m\u001b[39m training to \u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mepochs\u001b[39m}\u001b[39;00m\u001b[39m epochs is finished, nothing to resume.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m'\u001b[39m \\\n\u001b[0;32m    594\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mStart a new training without resuming, i.e. \u001b[39m\u001b[39m'\u001b[39m\u001b[39myolo train model=\u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39margs\u001b[39m.\u001b[39mmodel\u001b[39m}\u001b[39;00m\u001b[39m'\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m    595\u001b[0m     LOGGER\u001b[39m.\u001b[39minfo(\n\u001b[0;32m    596\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m'\u001b[39m\u001b[39mResuming training from \u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39margs\u001b[39m.\u001b[39mmodel\u001b[39m}\u001b[39;00m\u001b[39m from epoch \u001b[39m\u001b[39m{\u001b[39;00mstart_epoch\u001b[39m \u001b[39m\u001b[39m+\u001b[39m\u001b[39m \u001b[39m\u001b[39m1\u001b[39m\u001b[39m}\u001b[39;00m\u001b[39m to \u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mepochs\u001b[39m}\u001b[39;00m\u001b[39m total epochs\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[0;32m    597\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mepochs \u001b[39m<\u001b[39m start_epoch:\n",
      "\u001b[1;31mAssertionError\u001b[0m: yolov8n.pt training to 500 epochs is finished, nothing to resume.\nStart a new training without resuming, i.e. 'yolo train model=yolov8n.pt'"
     ]
    }
   ],
   "source": [
    "# resume\n",
    "results = model.train(data='datadet.yml', epochs=100, resume=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "tp2",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.17"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
